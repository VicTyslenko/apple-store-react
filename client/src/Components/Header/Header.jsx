import { Link } from 'react-router-dom';
import CartLogo from '../Icons/CartLogo'
import FavouriteLogo from '../Icons/FavouriteLogo'
import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import './Header.scss'

const Header = () => {

    const cartCount = useSelector(state => state.cart.cartToLocal)
    const favCount = useSelector(state => state.favourite.favouriteToLocal)



    useEffect(() => {
        localStorage.setItem("cart", JSON.stringify(cartCount))
        localStorage.setItem("favourite", JSON.stringify(favCount))

    }, [cartCount, favCount])
    return (
        <>

            <div className="header">
                <Link to="/" style={{ textDecoration: 'none' }}> <h1 className='devices'>Apple store</h1></Link>

                <div className='header-svg-wrapp'>
                    <sup className='counter'>{favCount.length}</sup>
                    <Link to="/favourites">
                        <FavouriteLogo />
                    </Link>
                    <sup className='counter'>{cartCount.length}</sup>
                    <Link to="/cartitems">
                        <CartLogo />
                    </Link>


                </div>

            </div>
        </>
    )
}

export default Header






