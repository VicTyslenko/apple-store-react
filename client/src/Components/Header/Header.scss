.header {
    display: flex;
    justify-content: space-between;
    padding: 0 80px;
    background-color:#000000c7;
    border-radius: 10px;
    position: fixed;
    z-index: 100;
    top: 0;
    left: 0;
    right: 0;
}

.devices {
    color: white;
    text-transform: uppercase;
    font-weight: 600;
    font-size: 40px;
    font-family: 'Roboto', sans-serif;
}

sup {
    font-size: 20px;
    font-weight: 400;
    color: white;
}

.header-svg-wrapp {
    display: flex;
    align-items: center;
}

.svg-favourite {
    margin-right: 10px;
    cursor: pointer;
}

.red-cart {
    fill: #ff0000;
}

.white-cart {
    fill: #cbc9c9;
}