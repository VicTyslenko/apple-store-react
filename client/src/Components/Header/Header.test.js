import Header from "./Header"
import { render, screen } from "@testing-library/react"
import { Provider } from "react-redux"
import store from "../../store/store"
import { MemoryRouter } from "react-router-dom"
import '@testing-library/jest-dom'
describe('Header', () => {

    test("Header snapshot", () => {
        const header = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Header />
                </MemoryRouter>
            </Provider>)
        expect(header).toMatchSnapshot()
    });

    test("checking text", () => {
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <Header />
                </MemoryRouter>
            </Provider>
        );
        expect(screen.getByText("Devices list"))
    });
    test("checking class", () => {
        const { container } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Header />
                </MemoryRouter>
            </Provider>
        );
        const headerDescription = container.querySelector(".header")
        expect(headerDescription).toBeInTheDocument()
    })

})