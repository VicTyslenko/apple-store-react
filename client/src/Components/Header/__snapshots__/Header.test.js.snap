// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <a
          href="/"
          style="text-decoration: none;"
        >
           
          <h1
            class="devices"
          >
            Devices list
          </h1>
        </a>
        <div
          class="header-svg-wrapp"
        >
          <sup
            class="counter"
          >
            0
          </sup>
          <a
            href="/favourites"
          >
            <svg
              class="svg-favourite-white"
              data-name="Layer 2"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M29.95 12.68A1 1 0 0 0 29 12h-9.26L17 2.77a1 1 0 0 0-1.91 0L12.26 12H3a1 1 0 0 0-.6 1.8l7.39 5.54-3.72 9.29a1 1 0 0 0 1.54 1.16L16 23.27l8.39 6.52a1 1 0 0 0 1.54-1.16l-3.72-9.29 7.39-5.54a1 1 0 0 0 .35-1.12Z"
              />
            </svg>
          </a>
          <sup
            class="counter"
          >
            0
          </sup>
          <a
            href="/cartitems"
          >
            <svg
              class="white-cart"
              data-name="Layer 2"
              height="32px"
              viewBox="0 0 35 35"
              width="32px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M27.47,23.93H14.92A5.09,5.09,0,0,1,10,20L8,11.87a5.11,5.11,0,0,1,5-6.32h16.5a5.11,5.11,0,0,1,5,6.32l-2,8.15A5.1,5.1,0,0,1,27.47,23.93ZM12.94,8.05a2.62,2.62,0,0,0-2.54,3.23l2,8.15a2.6,2.6,0,0,0,2.54,2H27.47a2.6,2.6,0,0,0,2.54-2l2-8.15a2.61,2.61,0,0,0-2.54-3.23Z"
              />
              <path
                d="M9.46 14a1.25 1.25 0 0 1-1.21-1L6.46 5.23A3.21 3.21 0 0 0 3.32 2.75H1.69a1.25 1.25 0 0 1 0-2.5H3.32A5.71 5.71 0 0 1 8.9 4.66l1.78 7.77a1.24 1.24 0 0 1-.93 1.5A1.43 1.43 0 0 1 9.46 14zM15.11 34.75a4 4 0 1 1 4-4A4 4 0 0 1 15.11 34.75zm0-5.54a1.52 1.52 0 1 0 1.52 1.52A1.52 1.52 0 0 0 15.11 29.21zM28.93 34.75a4 4 0 1 1 4-4A4 4 0 0 1 28.93 34.75zm0-5.54a1.52 1.52 0 1 0 1.53 1.52A1.52 1.52 0 0 0 28.93 29.21z"
              />
              <path
                d="M28.93,29.21H12.27a3.89,3.89,0,1,1,0-7.78h2.65a1.25,1.25,0,1,1,0,2.5H12.27a1.39,1.39,0,1,0,0,2.78H28.93a1.25,1.25,0,0,1,0,2.5Z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <a
        href="/"
        style="text-decoration: none;"
      >
         
        <h1
          class="devices"
        >
          Devices list
        </h1>
      </a>
      <div
        class="header-svg-wrapp"
      >
        <sup
          class="counter"
        >
          0
        </sup>
        <a
          href="/favourites"
        >
          <svg
            class="svg-favourite-white"
            data-name="Layer 2"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M29.95 12.68A1 1 0 0 0 29 12h-9.26L17 2.77a1 1 0 0 0-1.91 0L12.26 12H3a1 1 0 0 0-.6 1.8l7.39 5.54-3.72 9.29a1 1 0 0 0 1.54 1.16L16 23.27l8.39 6.52a1 1 0 0 0 1.54-1.16l-3.72-9.29 7.39-5.54a1 1 0 0 0 .35-1.12Z"
            />
          </svg>
        </a>
        <sup
          class="counter"
        >
          0
        </sup>
        <a
          href="/cartitems"
        >
          <svg
            class="white-cart"
            data-name="Layer 2"
            height="32px"
            viewBox="0 0 35 35"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M27.47,23.93H14.92A5.09,5.09,0,0,1,10,20L8,11.87a5.11,5.11,0,0,1,5-6.32h16.5a5.11,5.11,0,0,1,5,6.32l-2,8.15A5.1,5.1,0,0,1,27.47,23.93ZM12.94,8.05a2.62,2.62,0,0,0-2.54,3.23l2,8.15a2.6,2.6,0,0,0,2.54,2H27.47a2.6,2.6,0,0,0,2.54-2l2-8.15a2.61,2.61,0,0,0-2.54-3.23Z"
            />
            <path
              d="M9.46 14a1.25 1.25 0 0 1-1.21-1L6.46 5.23A3.21 3.21 0 0 0 3.32 2.75H1.69a1.25 1.25 0 0 1 0-2.5H3.32A5.71 5.71 0 0 1 8.9 4.66l1.78 7.77a1.24 1.24 0 0 1-.93 1.5A1.43 1.43 0 0 1 9.46 14zM15.11 34.75a4 4 0 1 1 4-4A4 4 0 0 1 15.11 34.75zm0-5.54a1.52 1.52 0 1 0 1.52 1.52A1.52 1.52 0 0 0 15.11 29.21zM28.93 34.75a4 4 0 1 1 4-4A4 4 0 0 1 28.93 34.75zm0-5.54a1.52 1.52 0 1 0 1.53 1.52A1.52 1.52 0 0 0 28.93 29.21z"
            />
            <path
              d="M28.93,29.21H12.27a3.89,3.89,0,1,1,0-7.78h2.65a1.25,1.25,0,1,1,0,2.5H12.27a1.39,1.39,0,1,0,0,2.78H28.93a1.25,1.25,0,0,1,0,2.5Z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
